packages/react-fiber-react-18/src/index.js

```diff
@@ -3,9 +3,9 @@ import { makeHostConfig } from './hostconfig';
 import rootPackageJson from '../../../package.json';
 import packageJson from '../package.json';
 
-export const REACT_DOM_VERSION = rootPackageJson.dependencies[
+export const REACT_DOM_VERSION = rootPackageJson?.dependencies?.[
     'react-dom'
-].replace(/[^0-9.]/g, '');
+]?.replace(/[^0-9.]/g, '') || '18.0.0';
```

Comment:
I dont think this code will work outside of the monorepo as `rootPackageJson` may not exist?

Seems like this could cause issues. Is there another way of getting this information?

Reply:
Not directly related to this line, but to the whole file.

1. I am pretty sure `version` passed to `injectIntoDevTools` doesn't need to be the version of ReactDOM any more. See: 
1.1. https://github.com/pmndrs/react-three-fiber/pull/1231
1.2. https://github.com/facebook/react/pull/21268
1.3. https://github.com/facebook/react/pull/21269
2. Please check if the `package.json` file imports are removed from the bundle. From my experience with rollup when I encountered this issue they were not. This increased bundle size unnecessarily. We can provide values read from package json file with the `replace` rollup plugin similar to `__DEV__` value.

Reply:
@Zyie does the code need to work outside of a monorepo? There's already quite a few things that depend on it being inside a monorepo - eslint, jest config and rollup config to name a few. I agree this code is nasty though. We could install React inside just this package, but because almost all the packages depend on it (as with other shared dependencies) I've listed them in the root `package.json` to keep versioning easier to manage.

@michalochman 

1. From your links It appears that the `version` needs to be the current "React version" whether from DOM or React.
2. I can confirm that right now the whole `package.json` file was being bundled into the build. But in my Typescript PR, it wouldn't let me do an import from outside the package anyway, so I've already written a script to handle this situation and write a json file inside the src dir: https://github.com/pixijs/pixi-react/pull/412/files#diff-68eaf1ddcab994998bf4397fbcd8f7685590ac3760239880627679d5141c3ff4